tests: pwm: led: Fix pwm_led sample

PerMac opened the pull request:
The PR fixes the pwm_led sample:
- now test doesn't rely on any specific device name
- Logs which are scanned with regex in the test are printed
  only after a command passes (before failures were not
  affecting the test verdict)
- If 1 sec cycle is not supported appropriate info is printed
  and won't cause the test to failed
- Changed second "Turned off" msg so regex doesn't mix it with the
  the first one.

Fixes #35524

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

PerMac committed:
tests: pwm: led: Fix pwm_led sample

The PR fixes the pwm_led sample:
- now test doesn't rely on any specific device name
- Logs which are scanned with regex in the test are printed
  only after a command passes (before failures were not
  affecting the test verdict)
- If 1 sec cycle is not supported appropriate info is printed
  and won't cause the test to failed
- Changed second "Turned off" msg so regex doesn't mix it with the
  the first one.

Fixes #35524

Signed-off-by: Maciej Perkowski <Maciej.Perkowski@nordicsemi.no>

anangl commented on samples/drivers/led_pwm/sample.yaml:
```diff
@@ -13,11 +13,10 @@ tests:
       ordered: true
       regex:
-        - "Found device LED_PWM_\\d+"
+        - "Found device [^\\s]"
```
```suggestion
        - "Found device \\S+"
```
To match one or more non-whitespace characters at the end of the line.

anangl commented on samples/drivers/led_pwm/src/main.c:
```diff
@@ -48,24 +48,23 @@ static void run_led_test(const struct device *led_pwm, uint8_t led)
 	k_sleep(K_MSEC(1000));
 
 	/* Increase LED brightness gradually up to the maximum level. */
-	LOG_INF("  Increasing brightness gradually");
```
In this case, I think that message was more appropriate than "Increased" printed after the loop, when the app is no longer presenting anything interesting.

simonguinot commented on samples/drivers/led_pwm/src/main.c:
```diff
@@ -74,33 +73,25 @@ static void run_led_test(const struct device *led_pwm, uint8_t led)
-	if (err < 0) {
-		LOG_ERR("err=%d", err);
-		return;
-	}
```
Hi @PerMac.

Please, keep this test.

Configuring the PWM for a LED is a trade-off. You want the PWM frequency as high as possible to fool eyes and camera on steady brightness levels. But on the other end, you also want the PWM configuration to support a 0.5 Hz hardware blinking which is a pretty common need. This test allows to verify that.

Concretely, on a ST board, this helps to check that the value assignated to "st,prescaler" DT property is correct.

simonguinot commented on samples/drivers/led_pwm/src/main.c:
```diff
@@ -74,33 +73,25 @@ static void run_led_test(const struct device *led_pwm, uint8_t led)
 		}
 		k_sleep(FADE_DELAY);
 	}
+	LOG_INF("  Increased brightness gradually");
```
So now we get something useful / parsable for a script, but useless for a human :)

With this message printed after the test, it will be difficult for a human to be sure that the LED behaved correctly.

PerMac commented:
@anangl @simonguinot Thanks for you commments. Indeed, I focused on the test's point of view and forgot about the user experience. I believe I addressed all your comments. I brought 1 sec period back and added else case (and regex check) when such period is not supported.
In addition, I changed the log/check for the last "turned off" as it was possible it got mixed with the one from the beginning of the loop.

simonguinot commented:
Thanks for this patch @PerMac.
